stats/src/main/java/io/airlift/stats/cardinality/BucketListener.java

```diff
@@ -0,0 +1,19 @@
+/*
```

Comment:
Do you have results from before?

Reply:
Before, we'd convert the sparse HLL into a dense HLL and then merge it. So the performance is approximately what the first benchmark showed (there's some extra cost in the conversion that we're not measuring):

`21.557 ± 1.116 us/op` for current implementation,
`11.677 ± 0.788  us/op` after Dense HLL merging is optimized,
`0.320 ± 0.013 us/op` after dedicated merge implementation for Dense vs Sparse